### ISSUE TITLE
Material Generation Supported?

### ISSUE THREAD

**Issue by Sheldonmao** [nv-tlabs/GET3D#42]
Hi, I am interested in the material generation ability described in the paper. However, I can't find relevant code relating to either rendering images using material properties or generating real images from Turbosquid models (with consistent materials) using Blender. 
Did I miss something in the repo? Or the material generation code is not available yet?

**Comment by SteveJunGao** [comment]
Hi @Sheldonmao,

We haven't included that part in the code yet, this is in our future plan of release, please stay tuned!

**Comment by AmanKishore** [comment]
Hi @Sheldonmao you should try out Texture Generation on https://app.mirageml.com/textures
Would love to chat when you're available!

**Comment by Siziff** [comment]
I read the article very carefully and I can't understand, the blender for training generates 24 views from different sides of the 3d figure, but where are these 24 views served in the model? at what point is the model being trained to adjust to these types to generate a distribution and generate other similar f shapes?

> Hi @Sheldonmao,
> 
> We haven't included that part in the code yet, this is in our future plan of release, please stay tuned!

**Comment by SteveJunGao** [comment]
Hi, @Siziff Thanks for checking our paper so carefully!

Re 1: the rendered images from the synthetic dataset (e.g. shapenet) are used to feed into the discriminator to classify whether they're real or fake (Eq. 1, $p_x$ is the distribution of real image). The loss function from the discriminator can be backpropagated to the generator to generate the desired 3d shapes that resemble the training data. 

Re 2: Yes, you're almost correct. $h_i * w_i,j,k,l,m$ is from Eq. 4 in the Appendix, this is the modulated convolution kernel weights, after we do demodulation (Eq. 5), the weight will be the convolution kernel that goes through the  $F^{\prime}_{geo}$

Re 3: Sorry for the confusion here, we simply repeat the vector $h_i$ along the 2,3,4,5 dimensions to make it the same size as the matrix $ w_i,j,k,l,m$, you can check the code in this [line](https://github.com/nv-tlabs/GET3D/blob/master/training/geometry_predictor.py#L166).

**Comment by SteveJunGao** [comment]
Hi @Siziff , any further questions regarding this issue?

**Comment by SteveJunGao** [comment]
Close the issue as no updated received
